Mathlib/NumberTheory/Harmonic/ZetaAsymp.lean

```diff
@@ -0,0 +1,350 @@
+  rw [term_tsum_of_lt hs]
+  generalize (∑' n : ℕ, 1 / (n + 1 : ℝ) ^ s) = Z
+  field_simp [(show s - 1 ≠ 0 by linarith)]
+  ring_nf
+
+end s_gt_one
+
+section continuity
+
+/-!
+## Continuity of the sum
+-/
+
+lemma continuousOn_term (n : ℕ) :
+    ContinuousOn (fun x ↦ term (n + 1) x) (Ici 1) := by
+  -- TODO: can this be shortened using the lemma
+  -- `continuous_parametric_intervalIntegral_of_continuous'` from #11185?
+  simp only [term, intervalIntegral.integral_of_le (by linarith : (↑(n + 1) : ℝ) ≤ ↑(n + 1) + 1)]
+  apply continuousOn_of_dominated (bound := fun x ↦ (x - ↑(n + 1)) / x ^ (2 : ℝ))
+  · exact fun s hs ↦ (term_welldef (by simp) (zero_lt_one.trans_le hs)).1.1
+  · intro s (hs : 1 ≤ s)
+    rw [ae_restrict_iff' measurableSet_Ioc]
+    filter_upwards with x hx
+    have : 0 < x := lt_trans (by positivity) hx.1
+    rw [norm_of_nonneg (div_nonneg (sub_nonneg.mpr hx.1.le) (by positivity)), Nat.cast_add_one]
+    apply div_le_div_of_nonneg_left
+    · exact_mod_cast sub_nonneg.mpr hx.1.le
+    · positivity
+    · exact rpow_le_rpow_of_exponent_le (le_trans (by simp) hx.1.le) (by linarith)
+  · rw [← IntegrableOn, ← intervalIntegrable_iff_integrableOn_Ioc_of_le (by linarith)]
+    exact_mod_cast term_welldef (by linarith : 0 < (n + 1)) zero_lt_one
+  · rw [ae_restrict_iff' measurableSet_Ioc]
+    filter_upwards with x hx
+    refine ContinuousAt.continuousOn (fun s (hs : 1 ≤ s) ↦ continuousAt_const.div ?_ ?_)
+    · exact continuousAt_const.rpow (continuousAt_id.add continuousAt_const) (Or.inr (by linarith))
+    · exact (rpow_pos_of_pos ((Nat.cast_pos.mpr (by simp)).trans hx.1) _).ne'
+
+lemma continuousOn_term_tsum : ContinuousOn term_tsum (Ici 1) := by
+  -- We use dominated convergence, using `fun n ↦ term n 1` as our uniform bound (since `term` is
+  -- monotone decreasing in `s`.)
+  refine continuousOn_tsum (fun i ↦ continuousOn_term _) term_tsum_one.summable (fun n s hs ↦ ?_)
+  rw [term, term, norm_of_nonneg]
+  · simp_rw [intervalIntegral.integral_of_le (by linarith : (↑(n + 1) : ℝ) ≤ ↑(n + 1) + 1)]
+    refine set_integral_mono_on ?_ ?_ measurableSet_Ioc (fun x hx ↦ ?_)
+    · exact (term_welldef n.succ_pos (zero_lt_one.trans_le hs)).1
+    · exact (term_welldef n.succ_pos zero_lt_one).1
+    · rw [div_le_div_left] -- leave side-goals to end and kill them all together
+      apply rpow_le_rpow_of_exponent_le
+      · exact (lt_of_le_of_lt (by simp) hx.1).le
+      · linarith [mem_Ici.mp hs]
+      · linarith [hx.1]
+      all_goals apply rpow_pos_of_pos ((Nat.cast_nonneg _).trans_lt hx.1)
+  · rw [intervalIntegral.integral_of_le (by linarith)]
+    refine set_integral_nonneg measurableSet_Ioc (fun x hx ↦ div_nonneg ?_ (rpow_nonneg ?_ _))
+    all_goals linarith [hx.1]
+
+/-- First version of the limit formula, with a limit over real numbers tending to 1 from above. -/
+lemma tendsto_riemannZeta_nhds_right : Tendsto (fun s : ℝ ↦ riemannZeta s - 1 / (s - 1))
```

Comment:
Maybe add a `sub_one_div` or so in the name (it's not asymptotics of the zeta function itself...)?

Reply:
Good point, thanks!